sympy/solvers/tests/test_ode.py

```diff
@@ -3705,3 +3705,29 @@ def test_issue_17322():
     sol = [Eq(f(x), C1), Eq(f(x), C1*exp(-x))]
     assert set(sol) == set(dsolve(eq, hint='lie_group'))
     assert checkodesol(eq, sol) == 2*[(True, 0)]
+
+def test_2nd_2F1_hypergeometric():
+
+    a, b, c = symbols("a b c")
+
+    eq = x*(x-1)*f(x).diff(x, 2) + ((a+b+1)*x-c)*f(x).diff(x) + a*b*f(x)
+    sol = Eq(f(x), C1*x**(1 - c)*hyper((a - c + 1, b - c + 1), (2 - c,), x) + C2*hyper((a, b), (c,), x))
+    assert sol == dsolve(eq, hint='2nd_hypergeometric')
+    # assert checkodesol(eq, sol) == (True, 0) Hanging
```

Comment:
Are we sure that this solution is correct?

Reply:
yes